static/js/jira-manual-app-creation.js

```diff
@@ -0,0 +1,28 @@
+$("#jiraManualAppCreation__form").submit((event) => {
+  event.preventDefault();
+
+  // TODO: Form submission
```

Comment:
I would expect the frontend validation to be a part of this PR but happy for you to tackle it separately if you prefer. We obviously won't be making a request to the backend to save the data in the db but we should at least take the data in and make sure all required fields (which is all of them) have been provided and render the `this is a required field message` for any empty fields.

Reply:
The form does include a `required` validation, using the default HTML `required` attribute. Do we need to do other validations too?

Reply:
Sorry, realised I replied in the wrong thread. https://github.com/atlassian/github-for-jira/pull/1331/files#r913338483

You can't just rely on the browser to handle this because the styling won't align with design. First of all, it looks nothing like what we have in Figma

<img width="649" alt="Screen Shot 2022-07-05 at 4 35 54 pm" src="https://user-images.githubusercontent.com/37155488/177265008-a7e4ee1f-b0cd-46e9-a626-0be8532f4b25.png">

and secondly, it doesn't flag **all** empty fields as required at the same time. If I miss multiple it will flag the first empty one. I then fill then in and when I go to submit it flags the next one. This is a pretty frustrating user experience.

<img width="609" alt="Screen Shot 2022-07-05 at 4 36 03 pm" src="https://user-images.githubusercontent.com/37155488/177265256-1ae354df-3654-411e-9dfb-8b36f85721ad.png">

It also behaves differently in each browser. In the video below, watch what happens in Safari. I click next and see the message at the top but then it's gone when I scroll up. Sure, the input field is highlighted but what if I didn't read the message?

https://user-images.githubusercontent.com/37155488/177266409-3f3a2a07-f7cc-4a2f-95c8-0b8b91198766.mov

[ ] Ditch `required` from all the input fields and handle the frontend check for that yourself (like you did on the server app url screen)

Reply:
Used [AUI Form validation](https://aui.atlassian.com/aui/latest/docs/form-validation.html) to match the design.

Reply:
We have a ticket to implement manual app creation on the backend which will include adding the ajax POST request, which will account for the `// TODO: Form submission` you have here. However, I would expect the payload that needs to be passed to be done as part of this PR. You should be pulling in all the data from the form and the object that will be passed to the backend.

Reply:
The data has been added.